app/actions/remote/custom_profile.ts

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {forceLogoutIfNecessary} from '@actions/remote/session';
+import DatabaseManager from '@database/manager';
+import NetworkManager from '@managers/network_manager';
+import {customProfileAttributeId} from '@utils/custom_profile_attribute';
+import {getFullErrorMessage} from '@utils/errors';
+import {logDebug} from '@utils/log';
+
+import type Model from '@nozbe/watermelondb/Model';
+import type {CustomProfileField, CustomAttribute, CustomAttributeSet, CustomProfileAttribute, UserCustomProfileAttributeSimple} from '@typings/api/custom_profile_attributes';
+
+/**
+ * Fetches custom profile attribute values for a user
+ * @param serverUrl - The server URL
+ * @param userId - The user ID
+ * @param filterEmpty - Whether to filter out empty values
+ * @returns Promise with the custom profile attributes or error
+ */
+export const fetchCustomProfileAttributes = async (serverUrl: string, userId: string, filterEmpty = false): Promise<{attributes: CustomAttributeSet; error: unknown}> => {
```

Comment:
The WS events do take care of removing fields and attributes that have been deleted, but this fetch action does NOT, that is totally missing here